pyhanko/sign/fields.py

```diff
@@ -1665,6 +1665,23 @@ def append_signature_field(
 
     if sig_field_spec.box is not None:
         llx, lly, urx, ury = sig_field_spec.box
+        matrix = None
+        if sig_field_spec.visible_sig_settings.rotate_with_page == False:
+            pagetree_obj = page_ref.get_object()
+            while '/Parent' in pagetree_obj and ('/Type' not in pagetree_obj or pagetree_obj['/Type'] != '/Page'):
+                pagetree_obj = pagetree_obj['/Parent']
+            if '/Rotate' in pagetree_obj:
+                match pagetree_obj['/Rotate']:
```

Comment:
Case matching is unfortunately not an option since pyHanko still supports Python 3.7+ ;)

Reply:
Changed it to if/elif